xbmc/settings/SettingsValueXmlSerializer.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (C) 2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "SettingsValueXmlSerializer.h"
+
+#include "settings/lib/SettingDefinitions.h"
+#include "settings/lib/SettingsManager.h"
+#include "settings/lib/SettingSection.h"
+#include "utils/XBMCTinyXML.h"
+#include "utils/XMLUtils.h"
+#include "utils/log.h"
+
+static constexpr char SETTINGS_XML_ROOT[] = "settings";
+
+std::string CSettingsValueXmlSerializer::SerializeValues(const CSettingsManager* settingsManager) const
+{
+  if (settingsManager == nullptr)
+    return "";
+
+  CXBMCTinyXML xmlDoc;
+  TiXmlElement rootElement(SETTINGS_XML_ROOT);
+  rootElement.SetAttribute(SETTING_XML_ROOT_VERSION, settingsManager->GetVersion());
+  TiXmlNode* xmlRoot = xmlDoc.InsertEndChild(rootElement);
+  if (xmlRoot == nullptr)
+    return "";
+
+  const auto sections = settingsManager->GetSections();
+  for (const auto& section : sections)
+    SerializeSection(xmlRoot, section);
+
+  std::stringstream stream;
+  stream << *xmlDoc.RootElement();
+
+  return stream.str();
+}
+
+void CSettingsValueXmlSerializer::SerializeSection(TiXmlNode* parent, std::shared_ptr<CSettingSection> section) const
```

Comment:
> The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

Reply:
Fixed in #17062.